arkouda/dataframe.py

```diff
@@ -56,37 +58,186 @@ def groupby_operators(cls):
                     )
+                elif isinstance(self.gb_key_names, list) and len(self.gb_key_names) == 1:
+                    return DataFrame(
+                        {c: self.gb.aggregate(self.df.data[c], opname)[1] for c in colnames},
+                        index=Index(self.gb.unique_keys, name=self.gb_key_names[0]),
+                    )
+                elif isinstance(self.gb_key_names, list):
+                    column_dict = {
+                        self.gb_key_names[i]: self.unique_keys[i] for i in range(len(self.gb_key_names))
+                    }
+                    for c in colnames:
+                        column_dict[c] = self.gb.aggregate(self.df.data[c], opname)[1]
+                    return DataFrame(column_dict)
+                else:
+                    return None
 
         return aggop
 
-    def count(self):
-        return Series(self.gb.count())
+    def count(self, as_series=None):
+        """
+        Compute the count of each value as the total number of rows, including NaN values.
+        This is an alias for size(), and may change in the future.
+
+        Parameters
+        ----------
+
+        as_series : bool, default=None
+            Indicates whether to return arkouda.dataframe.DataFrame (if as_series = False) or
+            arkouda.series.Series (if as_series = True)
+
+        Returns
+        -------
+
+        arkouda.dataframe.DataFrame (if as_series = False) or
+        arkouda.series.Series (if as_series = True)
+
+        """
+        if as_series is True or (as_series is None and self.as_index is True):
+            return self.__return_agg_series(self.gb.count())
+        else:
+            return self.__return_agg_dataframe(self.gb.count(), "count")
+
+    def size(self, as_series=None):
+        """
+        Compute the size of each value as the total number of rows, including NaN values.
+
+        Parameters
+        ----------
+
+        as_series : bool, default=None
+            Indicates whether to return arkouda.dataframe.DataFrame (if as_series = False) or
+            arkouda.series.Series (if as_series = True)
+
+        Returns
+        -------
+
+        arkouda.dataframe.DataFrame or
+        arkouda.series.Series
+
+        """
+        if as_series is True or (as_series is None and self.as_index is True):
+            return self.__return_agg_series(self.gb.size())
+        else:
+            return self.__return_agg_dataframe(self.gb.size(), "size")
+
+    def __return_agg_series(self, values):
+        if self.as_index:
+            if isinstance(self.gb_key_names, str):
+                return Series(values, index=Index(self.gb.unique_keys, name=self.gb_key_names))
+            elif len(self.gb_key_names) == 1:
+                return Series(values, index=Index(self.gb.unique_keys, name=self.gb_key_names[0]))
+            else:
+                return Series(values)
+        else:
+            return Series(values)
```

Comment:
i think this could be simplified a bit
```python
        if self.as_index:
            if isinstance(self.gb_key_names, str):
                return Series(values, index=Index(self.gb.unique_keys, name=self.gb_key_names))
            elif len(self.gb_key_names) == 1:
                return Series(values, index=Index(self.gb.unique_keys, name=self.gb_key_names[0]))
        return Series(values)
```

Reply:
That is cleaner.  I made the change.  Thanks!